lib/src/source/hosted.dart

```diff
@@ -1100,3 +1113,97 @@ class _RefAndCache {
   @override
   bool operator ==(Object other) => other is _RefAndCache && other.ref == ref;
 }
+
+extension GCSChecksumValidation on http.StreamedResponse {
+  static const checksumsHeaderName = 'x-goog-hash';
+
+  bool get isFromGCS {
```

Comment:
I don't think we have to mention GCS, we always check the hash if the header is present.